src/main/java/org/folio/service/pieces/flows/delete/PieceDeleteFlowManager.java

```diff
@@ -1,59 +1,42 @@
 package org.folio.service.pieces.flows.delete;
 
 import static org.folio.orders.utils.ProtectedOperationType.DELETE;
-import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS;
-import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS_NAME;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
-
-import org.apache.commons.collections4.CollectionUtils;
-import org.apache.commons.lang3.StringUtils;
-import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.BooleanUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.folio.models.ItemStatus;
 import org.folio.models.pieces.PieceDeletionHolder;
 import org.folio.rest.RestConstants;
 import org.folio.rest.core.exceptions.ErrorCodes;
 import org.folio.rest.core.exceptions.HttpException;
 import org.folio.rest.core.models.RequestContext;
-import org.folio.rest.jaxrs.model.Error;
-import org.folio.rest.jaxrs.model.Errors;
-import org.folio.rest.jaxrs.model.Piece;
 import org.folio.service.CirculationRequestsRetriever;
 import org.folio.service.ProtectionService;
-import org.folio.service.inventory.InventoryItemManager;
 import org.folio.service.pieces.PieceStorageService;
-import org.folio.service.pieces.PieceUpdateInventoryService;
 import org.folio.service.pieces.flows.BasePieceFlowHolderBuilder;
 
 import io.vertx.core.Future;
-import io.vertx.core.json.JsonObject;
 
 public class PieceDeleteFlowManager {
+
   private static final Logger logger = LogManager.getLogger(PieceDeleteFlowManager.class);
 
+  private final PieceDeleteFlowInventoryManager pieceDeleteFlowInventoryManager;
```

Comment:
It is a bit hard to understand when manager injects another manager. We can rename PieceDeleteFlowInventoryManager . We already have PieceUpdateInventoryService. Maybe it makes sense to rename to PieceDeleteInventoryService that will not have dependency to PieceUpdateInventoryService.
And PieceDeleteFlowManager will invoke both PieceUpdateInventoryService and PieceDeleteInventoryService

Reply:
Introduced the new service

Reply:
now contract is clear, thanks